### PR TITLE
Satellite ca support

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,8 +3,7 @@ module github.com/IBM/secret-common-lib
 go 1.16
 
 require (
-	github.com/IBM/secret-utils-lib v0.0.0-20220411122140-7d2c0d0e5d07
-	github.com/pierrre/gotestcover v0.0.0-20160517101806-924dca7d15f0 // indirect
+	github.com/IBM/secret-utils-lib v0.0.0-20220506111001-93b93a0e312f
```
Comment (arahamad) [inline go.mod:6]: can you create the release for secret-utils-lib repo and update the dependency here

File: pkg/secret_provider/unmanaged_secret_provider.go [hunk 2]
```diff
@@ -42,14 +49,37 @@ func newUnmanagedSecretProvider(logger *zap.Logger) (*UnmanagedSecretProvider, e
...
+
+	// Checking if the secret(api key) needs to be decoded
+	if authType == utils.DEFAULT && os.Getenv("IS_SATELLITE") == "True" {
+		logger.Info("Decoding apiKey since it's a satellite cluster")
+		decodedSecret, err := base64.StdEncoding.DecodeString(authenticator.GetSecret())
+		if err != nil {
+			logger.Error("Error decoding the secret", zap.Error(err))
+			return nil, err
+		}
+		authenticator.SetSecret(strings.TrimSuffix(string(decodedSecret), "\n"))
```
Comment (arahamad) [inline pkg/secret_provider/unmanaged_secret_provider.go:79]: can you add the comments why this trim is required

File: pkg/utils/errors.go [hunk 3]
```diff
@@ -17,7 +17,6 @@
 package utils
 
 const (
-	// TODO - Remove this
-	// ErrInitializingSecretProvider ...
-	ErrInitializingSecretProvider = "error initlializing secret provider"
+	// ErrDecryptionNotSupported ...
+	ErrDecryptionNotSupported = "API key is encrypted, decryption is supported only by sidecar"
```
Comment (arahamad) [inline pkg/utils/errors.go:21]: As per configuration API key is encrypted which is not supported by the `sss` library.

**Review by arahamad — CHANGES_REQUESTED** [review]
can you pl do these small changes

File: pkg/utils/errors.go (hunk 3 above)
Comment (GunaKKIBM) [reply pkg/utils/errors.go]: Updated with new error message.

File: pkg/secret_provider/unmanaged_secret_provider.go (hunk 2 above)
Comment (GunaKKIBM) [reply pkg/secret_provider/unmanaged_secret_provider.go:80]: Added comments.

File: go.mod (hunk 1 above)
Comment (GunaKKIBM) [reply go.mod]: Updated the library.

File: client/client.go [hunk 4]
```diff
@@ -7,7 +7,12 @@ import (
 )
 
 func main() {
-	secretprovider, _ := sp.NewSecretProvider()
+	secretprovider, err := sp.NewSecretProvider()
+	if err != nil {
+		fmt.Println("Error initializing provider")
+		fmt.Println(err)
```
Comment (prankulmahajan) [inline client/client.go:13]: Can you log this instead of printing it?

**Review by prankulmahajan — CHANGES_REQUESTED** [review]
Minor comment. Otherwise lgtm

**Review by prankulmahajan — APPROVED** [review]

**Review by sameshai — APPROVED** [review]
lgtm

**Review by arahamad — APPROVED** [review]
lgtm

**Pull request merged**